.github/workflows/lint.yml

```diff
@@ -81,27 +81,17 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-      - uses: Swatinem/rust-cache@v1
-        with:
-          # TODO: change to shared-key when we switch to Swatinem/rust-cache@v2
-          sharedKey: "clippy-cargo-lock"
-
-      - name: Run clippy action to produce annotations
-        uses: actions-rs/clippy-check@v1.0.7
-        if: ${{ steps.check_permissions.outputs.has-permission }}
-        with:
-          # GitHub displays the clippy job and its results as separate entries
-          name: Clippy (stable) Results
-          token: ${{ secrets.GITHUB_TOKEN }}
-          args: --all-features --all-targets -- -D warnings
```

Comment:
Please don't delete these clippy annotations, they are really useful for developers and reviewers to see where the clippy problems are in the GitHub diff.

Reply:
I'll leave the previous one as it's secure. Those are being updated with dependency bots, and some have changes that I can't deep dive into them right now to understand if the changes are "secure".

To be on the safe side, let's keep `actions-rs/clippy-check` and I'll keep this task as we have time (until September 27th) to make a better decision for this specific action